lib/dialogs/user_context.js

```diff
@@ -98,9 +99,15 @@ module.exports = async function resolveUserContext(dlg, variable) {
 
     let answer = await dlg.ask(type, question);
     if (saveToContext) {
-        var platform = dlg.manager.platform;
-        var sharedPrefs = platform.getSharedPreferences();
+        const platform = dlg.manager.platform;
+        const sharedPrefs = platform.getSharedPreferences();
 
+        if (type === ValueCategory.Location) {
+            const resolved = await lookupLocation(dlg, answer.value.name);
```

Comment:
You can't do this unconditionally - the answer might be an absolute location already, for example if you click on the location picker in Android (and in the tests).